### PR TITLE
Remove GetWindowLong and SetWindowLong from UnsafeNativeMethods.cs an…

### PR DESCRIPTION
**Pull request by gpetrou**
…d GWL constants from NativeMethods.cs

## Proposed changes

- Remove GetWindowLong and SetWindowLong from UnsafeNativeMethods.cs
- Remove GWL constants from NativeMethods.cs

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2447)

### CODE REVIEW

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs [hunk 1]
```diff
@@ -2449,7 +2449,7 @@ public void ShowDropDown(Control parent)
                 {
                     try
                     {
-                        UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(parent, parent.Handle));
+                        User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT, new HandleRef(parent, parent.Handle));
```
Comment (hughbe) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]: nit: use `IHandle` overloads

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs [hunk 2]
```diff
@@ -2463,8 +2463,7 @@ public void ShowDropDown(Control parent)
                     }
                     finally
                     {
-
-                        UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(null, IntPtr.Zero));
+                        User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT, new HandleRef(null, IntPtr.Zero));
```
Comment (hughbe) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionPanel.cs]: - get rid of `new HandleRef(null, IntPtr.Zero)` and turn into `IntPtr.Zero`
- use IHandle overload

i.e. -> `User32.SetWindowLong(this, User32.GWL.HWNDPARENT, IntPtr.Zero);`

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionUI.cs [hunk 3]
```diff
@@ -714,7 +714,7 @@ internal void ShowDesignerActionPanel(IComponent relatedComponent, DesignerActio
                 {
                     Debug.WriteLineIf(s_designeActionPanelTraceSwitch.TraceVerbose, "Assigning owner to mainParentWindow");
                     Debug.WriteLineIf(DesignerActionUI.DropDownVisibilityDebug.TraceVerbose, "Assigning owner to mainParentWindow");
-                    UnsafeNativeMethods.SetWindowLong(new HandleRef(designerActionHost, designerActionHost.Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(_mainParentWindow, _mainParentWindow.Handle));
+                    User32.SetWindowLong(new HandleRef(designerActionHost, designerActionHost.Handle), User32.GWL.HWNDPARENT, new HandleRef(_mainParentWindow, _mainParentWindow.Handle));
```
Comment (hughbe) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionUI.cs]: ditto regarding all the overloads where we can use `IHandle`

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.cs [hunk 4]
```diff
@@ -3576,7 +3576,7 @@ private void DetachWindow()
         private void InformOfNewHandle()
         {
             Debug.Assert(IsHandleCreated, "we got to have a handle to be here...");
-            wndprocAddr = UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_WNDPROC);
+            wndprocAddr = User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.WNDPROC);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/AxHost.cs]: Can this be?
```suggestion
            wndprocAddr = User32.GetWindowLong(this, User32.GWL.WNDPROC);
```

The same applies to other instances

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionUI.cs (hunk 3 above)
Comment (RussKie) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionUI.cs]: Stil outstanding

File: src/System.Windows.Forms/src/System/Windows/Forms/Application.ThreadContext.cs [hunk 5]
```diff
@@ -1075,7 +1075,7 @@ private unsafe void RunMessageLoopInner(msoloop reason, ApplicationContext conte
                     // If the owner window of the dialog is still enabled, disable it now.
                     // This can happen if the owner window is from a different thread or
                     // process.
-                    hwndOwner = UnsafeNativeMethods.GetWindowLong(new HandleRef(_currentForm, _currentForm.Handle), NativeMethods.GWL_HWNDPARENT);
+                    hwndOwner = User32.GetWindowLong(new HandleRef(_currentForm, _currentForm.Handle), User32.GWL.HWNDPARENT);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Application.ThreadContext.cs]: No need for `HandleRef` here

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 6]
```diff
@@ -3966,11 +3966,11 @@ private int WindowExStyle
         {
             get
             {
-                return unchecked((int)(long)UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_EXSTYLE));
+                return unchecked((int)(long)User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.EXSTYLE));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: No need for `HandleRef` here

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 7]
```diff
@@ -3966,11 +3966,11 @@ private int WindowExStyle
...
             }
             set
             {
-                UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_EXSTYLE, new HandleRef(null, (IntPtr)value));
+                User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.EXSTYLE, new HandleRef(null, (IntPtr)value));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 8]
```diff
@@ -3981,11 +3981,11 @@ internal int WindowStyle
         {
             get
             {
-                return unchecked((int)(long)UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_STYLE));
+                return unchecked((int)(long)User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.STYLE));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 9]
```diff
@@ -3981,11 +3981,11 @@ internal int WindowStyle
...
             }
             set
             {
-                UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_STYLE, new HandleRef(null, (IntPtr)value));
+                User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.STYLE, new HandleRef(null, (IntPtr)value));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.cs [hunk 10]
```diff
@@ -11353,8 +11353,8 @@ private void SetWindowFont()
 
         private void SetWindowStyle(int flag, bool value)
         {
-            int styleFlags = unchecked((int)((long)UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_STYLE)));
-            UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_STYLE, new HandleRef(null, (IntPtr)(value ? styleFlags | flag : styleFlags & ~flag)));
+            int styleFlags = unchecked((int)((long)User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.STYLE)));
+            User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.STYLE, new HandleRef(null, (IntPtr)(value ? styleFlags | flag : styleFlags & ~flag)));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 11]
```diff
@@ -1805,11 +1805,11 @@ public double Opacity
                     formState[FormStateLayered] = (TransparencyKey != Color.Empty) ? 1 : 0;
                     if (oldLayered != (formState[FormStateLayered] != 0))
                     {
-                        int exStyle = unchecked((int)(long)UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_EXSTYLE));
+                        int exStyle = unchecked((int)(long)User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.EXSTYLE));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 12]
```diff
@@ -1805,11 +1805,11 @@ public double Opacity
...
                         CreateParams cp = CreateParams;
                         if (exStyle != cp.ExStyle)
                         {
-                            UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_EXSTYLE, new HandleRef(null, (IntPtr)cp.ExStyle));
+                            User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.EXSTYLE, new HandleRef(null, (IntPtr)cp.ExStyle));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 13]
```diff
@@ -3279,7 +3279,7 @@ protected override void CreateHandle()
                 //
                 if (!ShowInTaskbar && OwnerInternal == null && TopLevel)
                 {
-                    UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, TaskbarOwner);
+                    User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT, TaskbarOwner);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 14]
```diff
@@ -3841,7 +3841,7 @@ protected void CenterToParent()
                 Size s = Size;
                 IntPtr ownerHandle = IntPtr.Zero;
 
-                ownerHandle = UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT);
+                ownerHandle = User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 15]
```diff
@@ -3897,7 +3897,7 @@ protected void CenterToScreen()
                 IntPtr hWndOwner = IntPtr.Zero;
                 if (TopLevel)
                 {
-                    hWndOwner = UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT);
+                    hWndOwner = User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 16]
```diff
@@ -5318,15 +5318,15 @@ public void Show(IWin32Window owner)
...
                 {
                     throw new ArgumentException(string.Format(SR.OwnsSelfOrOwner,
                                                       "show"), "owner");
                 }
 
                 // Set the new owner.
-                hWndOldOwner = UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT);
-                UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(owner, hWndOwner));
+                hWndOldOwner = User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT);
+                User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT, new HandleRef(owner, hWndOwner));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 17]
```diff
@@ -5432,15 +5432,15 @@ public DialogResult ShowDialog(IWin32Window owner)
...
                     {
                         throw new ArgumentException(string.Format(SR.OwnsSelfOrOwner,
                                                           "showDialog"), "owner");
                     }
 
                     // Set the new owner.
-                    hWndOldOwner = UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT);
-                    UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(owner, hWndOwner));
+                    hWndOldOwner = User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT);
+                    User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT, new HandleRef(owner, hWndOwner));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.cs [hunk 18]
```diff
@@ -5692,7 +5692,7 @@ private void UpdateHandleWithOwner()
                     }
                 }
 
-                UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, ownerHwnd);
+                User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT, ownerHwnd);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs [hunk 19]
```diff
@@ -707,7 +707,7 @@ protected override CreateParams CreateParams
                 //
                 if (IsHandleCreated)
                 {
-                    int currentStyle = unchecked((int)((long)UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_STYLE)));
+                    int currentStyle = unchecked((int)((long)User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.STYLE)));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs [hunk 20]
```diff
@@ -4525,9 +4525,9 @@ protected override void OnHandleCreated(EventArgs e)
             //
             if (!Scrollable)
             {
-                int style = unchecked((int)((long)UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_STYLE)));
+                int style = unchecked((int)((long)User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.STYLE)));
                 style |= NativeMethods.LVS_NOSCROLL;
-                UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_STYLE, new HandleRef(null, (IntPtr)style));
+                User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.STYLE, new HandleRef(null, (IntPtr)style));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListView.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs [hunk 21]
```diff
@@ -442,8 +442,8 @@ internal bool DrawValuesRightToLeft
             {
                 if (edit != null && edit.IsHandleCreated)
                 {
-                    int exStyle = unchecked((int)((long)UnsafeNativeMethods.GetWindowLong(new HandleRef(edit, edit.Handle), NativeMethods.GWL_EXSTYLE)));
-                    return ((exStyle & (int)User32.WS_EX.RTLREADING) != 0);
+                    int exStyle = unchecked((int)((long)User32.GetWindowLong(new HandleRef(edit, edit.Handle), User32.GWL.EXSTYLE)));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs [hunk 22]
```diff
@@ -1948,7 +1948,7 @@ public virtual int GetValueWidth()
             // control is a top=level window. standard way of setparent on the control is prohibited for top-level controls.
             // It is unknown why this control was created as a top-level control. Windows does not recommend this way of setting parent.
             // We are not touching this for this relase. We may revisit it in next release.
-            UnsafeNativeMethods.SetWindowLong(new HandleRef(dropDownHolder, dropDownHolder.Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(this, Handle));
+            User32.SetWindowLong(new HandleRef(dropDownHolder, dropDownHolder.Handle), User32.GWL.HWNDPARENT, new HandleRef(this, Handle));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/PropertyGridInternal/PropertyGridView.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDown.cs [hunk 23]
```diff
@@ -1697,7 +1697,7 @@ private void ResetDropShadowEnabled()
         private void ReparentToActiveToolStripWindow()
         {
             ToolStripManager.ModalMenuFilter.SetActiveToolStrip(this);
-            UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_HWNDPARENT, ToolStripManager.ModalMenuFilter.ActiveHwnd);
+            User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.HWNDPARENT, ToolStripManager.ModalMenuFilter.ActiveHwnd);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripDropDown.cs]: ditto, and below

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolTip.cs [hunk 24]
```diff
@@ -778,9 +778,9 @@ private void CreateHandle()
             // If in OwnerDraw mode, we don't want the default border.
             if (OwnerDraw)
             {
-                int style = unchecked((int)((long)UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_STYLE)));
+                int style = unchecked((int)((long)User32.GetWindowLong(this, User32.GWL.STYLE)));
                 style &= ~(int)User32.WS.BORDER;
-                UnsafeNativeMethods.SetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_STYLE, new HandleRef(null, (IntPtr)style));
+                User32.SetWindowLong(new HandleRef(this, Handle), User32.GWL.STYLE, new HandleRef(null, (IntPtr)style));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolTip.cs]: ditto

File: src/System.Windows.Forms/src/System/Windows/Forms/TreeView.cs [hunk 25]
```diff
@@ -2002,20 +2002,19 @@ protected override void OnHandleCreated(EventArgs e)
             // This seems to make the Treeview happy.
             if (CheckBoxes)
             {
-                int style = unchecked((int)(UnsafeNativeMethods.GetWindowLong(new HandleRef(this, Handle), NativeMethods.GWL_STYLE)));
+                int style = unchecked((int)User32.GetWindowLong(new HandleRef(this, Handle), User32.GWL.STYLE));
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/TreeView.cs]: ditto, and below

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionUI.cs [hunk 26]
```diff
@@ -714,7 +714,7 @@ internal void ShowDesignerActionPanel(IComponent relatedComponent, DesignerActio
                 {
                     Debug.WriteLineIf(s_designeActionPanelTraceSwitch.TraceVerbose, "Assigning owner to mainParentWindow");
                     Debug.WriteLineIf(DesignerActionUI.DropDownVisibilityDebug.TraceVerbose, "Assigning owner to mainParentWindow");
-                    UnsafeNativeMethods.SetWindowLong(new HandleRef(designerActionHost, designerActionHost.Handle), NativeMethods.GWL_HWNDPARENT, new HandleRef(_mainParentWindow, _mainParentWindow.Handle));
+                    User32.SetWindowLong(designerActionHost, User32.GWL.HWNDPARENT, new HandleRef(_mainParentWindow, _mainParentWindow.Handle));
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionUI.cs]: Why not?
```suggestion
                    User32.SetWindowLong(designerActionHost, User32.GWL.HWNDPARENT, _mainParentWindow));
```

Comment (gpetrou) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/DesignerActionUI.cs]: This is an IWin32Window.

**Pull request merged**